qa/workunits/rbd/rbd_mirror_helpers.sh

```diff
@@ -151,8 +176,10 @@ setup()
     trap cleanup INT TERM EXIT
 
     if [ -n "${RBD_MIRROR_TEMDIR}" ]; then
+	test -d "${RBD_MIRROR_TEMDIR}" ||
 	mkdir "${RBD_MIRROR_TEMDIR}"
 	TEMPDIR="${RBD_MIRROR_TEMDIR}"
+	cd ${TEMPDIR}
```

Comment:
Note, we also do `cd ${TEMPDIR}` below for `-z "${RBD_MIRROR_USE_EXISTING_CLUSTER}"` case. The mktemp case can be when we run on teuthlogy, and then I don't want to chdir.

When running tests locally I always want to chdir to tempdir (to locate clusters ceph.conf files properly). When running on teuthology I don.'t want to change to tempdir.

When RBD_MIRROR_TEMDIR is set, it means we run locally. When both RBD_MIRROR_TEMDIR and RBD_MIRROR_USE_EXISTING_CLUSTER are set, it means that we run locally but don't want to restart the clusters. When only RBD_MIRROR_USE_EXISTING_CLUSTER is set, it is supposed we run on teuthology.

I believe the code can be refactored if it still looks confusing to you.